README.md

```diff
@@ -176,3 +185,10 @@ All of these questions are in scope to be addressed in future discussions in the
 - What to do about e.g. `libsForQt5.callPackage`? This goes into overrides, a different problem to fix
 - What about aliases like `jami-daemon = jami.jami-daemon`?
 - What about `recurseIntoAttrs`? Not single packages, package sets, another problem
+
+# Definitions
+
+ - <a id="def-variant-attribute"/> *variant attribute*: an attribute that defines a package by invoking it with non-default arguments, for example:
+   ```
+     cairo_headless = cairo.override(o: { withGUI = false; });
```

Comment:
This would be allowed in any case, even with the tighter restriction. You can always reference attributes, the restriction is only for file references.

Reply:
Right, this is not the best example. Sometimes overlays and overrides don't interact the way we'd like them to.
See https://github.com/NixOS/nixpkgs/pull/204287/files#diff-ab5748dc9567516fefba8344056b51ec1866adeace380f46e58a7af3d619ea22

To summarize: if attributes are defined in terms of one another, an overlay can not define a variant to become the default, because that leads to a mutual recursion.